src/sql/widgets/table_widget/table_widget.py

```diff
@@ -0,0 +1,151 @@
+from sql.connection import Connection
+from IPython import get_ipython
+import math
+import sqlalchemy
+import time
+from sql.util import (
+    fetch_sql_with_pagination,
+    parse_sql_results_to_json,
+    is_table_exists,
+)
+
+from sql.widgets import utils
+from sql.telemetry import telemetry
+
+import os
+
+# Widget base dir
+BASE_DIR = os.path.dirname(__file__)
+
+
+class TableWidget:
+    @telemetry.log_call("TableWidget-init")
+    def __init__(self, table):
+        """
+        Creates an HTML table element and populates it with SQL table
+
+        Parameters
+        ----------
+        table : str
+            Table name where the data is located
+        """
+
+        self.html = ""
+
+        is_table_exists(table)
+
+        # load css
+        html_style = utils.load_css(f"{BASE_DIR}/css/tableWidget.css")
+        self.add_to_html(html_style)
+
+        self.create_table(table)
+
+        # register listener for jupyter lab
+        self.register_comm()
+
+        # load_tests
+        self.load_tests()
+
+    def _repr_html_(self):
+        return self.html
+
+    def add_to_html(self, html):
+        self.html += html
+
+    def create_table(self, table):
+        """
+        Creates an HTML table with default data
+        """
+        rows_per_page = 10
+        rows, columns = fetch_sql_with_pagination(table, 0, rows_per_page)
+        rows = parse_sql_results_to_json(rows, columns)
+
+        query = f"SELECT count(*) FROM {table}"
+        n_total = Connection.current.session.execute(
+            sqlalchemy.sql.text(query)
+        ).fetchone()[0]
+        table_name = table.strip('"').strip("'")
+
+        n_pages = math.ceil(n_total / rows_per_page)
+
+        unique_id = str(int(time.time()))
+        table_container_id = f"tableContainer_{unique_id}"
+
+        # Create table container with unique id
+        table_container_html = f"""
+            <div id="{table_container_id}" class="table-container"></div>
+            """
+        self.add_to_html(table_container_html)
+
+        html_scripts = utils.load_js(
+            [
+                f"{BASE_DIR}/js/tableWidget.js",
+                utils.set_template_params(
+                    columns=list(columns),
+                    rows_per_page=rows_per_page,
+                    n_pages=n_pages,
+                    n_total=n_total,
+                    table_name=table_name,
+                    table_container_id=table_container_id,
+                    table=table,
+                    initialRows=rows,
+                ),
+            ]
+        )
+        self.add_to_html(html_scripts)
+
+    def load_tests(self):
```

Comment:
what are these tests for?

Reply:
We need to initiate the `WidgetTable` instance to use this widget, then we can display it in the notebook.

e.g.

```python
from sql.widgets import TableWidget
from IPython.display import display

table_widget = TableWidget("my_table_name")
display(table_widget)
```

In the background, `WidgetTable` uses JS to draw this table dynamically and add it to the DOM. We import this JS directly from [`tableWidget.js`](https://github.com/yafimvo/jupysql/blob/table_explorer/src/sql/widgets/table_widget/js/tableWidget.js). This makes it difficult to test the JS behavior for 2 main reasons:
1. We need DOM
2. We probably want to test specific JS functionality

That's why we load the JS functions we want to test, so we can easily test them as part of `TableWidget` test unit

```python
def test_table_widget_create_table_rows(ip, rows, expected):
    table_widget = TableWidget("empty_table")

    create_table_rows = js2py.eval_js(table_widget.tests["createTableRows"])

    table_rows = create_table_rows(rows)

    assert table_rows == expected
```

The function we are testing:

```javascript
function createTableRows(rows) {
    const _html = rows.map(function(row) {
        const tds =
        Object.keys(row).map(function(key) {

            return "<td>" + row[key] + "</td>"
        }).join("") ;
        return "<tr>" + tds + "</tr>";
    }).join("");

    return _html
}
```

This allows us to:
1. Avoid DOM dependency
2. Test any JS function we want
3. Control `TableWidget` testing

Reply:
ok, please add some comment in the source code to explain what the load_tests function is doing